modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -247,6 +247,7 @@ public ObjectEntry addObjectEntry(
 
 		_setExternalReferenceCode(objectEntry, values);
 
+		objectEntry.setExternalReferenceCode(objectEntry.getUuid());
```

Comment:
The user can set a value for **externalReferenceCode** in some ObjectEntry, via API. With that changes, you are override the value of **externalReferenceCode** setted by the user, in this context.
Move this line to the above method `setExternalReferenceCode(...)` and make sure it just will be executed if **values** does not bring "externalReferenceCode" field